src/hotspot/cpu/aarch64/templateInterpreterGenerator_aarch64.cpp

```diff
@@ -1978,21 +1978,42 @@ address TemplateInterpreterGenerator::generate_trace_code(TosState state) {
 }
 
 void TemplateInterpreterGenerator::count_bytecode() {
-  Register rscratch3 = r0;
-  __ push(rscratch1);
-  __ push(rscratch2);
-  __ push(rscratch3);
+  Register rscratch3 = r10;
```

Comment:
Please pass the scratch register to use as an argument to `TemplateInterpreterGenerator::generate_trace_code`

Reply:
Thanks for your review. But I'm afraid I didn't fully understand it.

Why `generate_trace_code` is involved? I guess you mean `count_bytecode()`?
But `count_bytecode()` is invoked [here](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/interpreter/templateInterpreterGenerator.cpp#L362), and I don't think it's a proper site to pass arch-specific register `r10` to the general `count_bytecode()`.

Please correct me if I misunderstood. Thanks.

Reply:
Ah, I see what you mean.
OK, just delete the name `rscratch3` and pass `r10` to `atomic_addw`. That simplifies the code to good effect.

Reply:
Yes. Agree. Updated.